### DISCUSSION TITLE
Methods to handle continental size mosaics

### DISCUSSION THREAD

**Discussion by charalamm** [developmentseed/eoAPI#110] [General]
We are having a use case where we want to create and display a mosaic of the whole Africa at once and as far as I know eoAPI does not work well when it has to handle huge mosaics so I would like to open a discussion on possible solutions.

As a possible solution I am thinking of creating two collections for the same dataset:
* One collection containing the original high resolution raster files and having the purpose of being displayed in high zoom on the map
* The other collection containing processed rasters with low resolution and being displayed in low zoom levels

However this method would have the drawbacks:
* The frontend will need to use a different endpoint for different zoom levels
* There is a collection that has no usage other than demonstrative purposes
* Some resources are needed to calculate and save the low resolution rasters
* There is a dependency between the collections that might cause issues or be forgotten if or when updating the original collection

**Comment by vincentsarago** [comment]
@charalamm this is, sadly, a common issues where there is not a one good answer. 

The raster service will be limited by the data resolution/size, the smaller your data is, the more files you'll need to `open` when creating low zoom level tiles. 

To `resolve` this you could:
- change your input data to use `larger` COGs 
- create another level of COGs (as you proposed) and have your frontend switch between the two mosaics based on zoom level  

links:
- https://github.com/developmentseed/titiler/discussions/287#discussioncomment-579936
- https://github.com/developmentseed/mosaicjson-spec/discussions/22
